packages/core/admin/admin/src/utils/getFetchClient.ts

```diff
@@ -111,20 +91,133 @@ type FetchClient = {
  * };
  * ```
  */
-const getFetchClient = (defaultOptions: AxiosRequestConfig = {}): FetchClient => {
-  instance.defaults.baseURL = window.strapi.backendURL;
-  return {
-    get: (url, config) =>
-      instance.get(normalizeUrl(url), {
-        ...defaultOptions,
-        ...config,
-      }),
-    put: (url, data, config) =>
-      instance.put(normalizeUrl(url), data, { ...defaultOptions, ...config }),
-    post: (url, data, config) =>
-      instance.post(normalizeUrl(url), data, { ...defaultOptions, ...config }),
-    del: (url, config) => instance.delete(normalizeUrl(url), { ...defaultOptions, ...config }),
+const getFetchClient = (defaultOptions: FetchConfig = {}): FetchClient => {
+  const { options } = defaultOptions;
+  const headers = new Headers({
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+    Authorization: `Bearer ${getToken()}`,
+  });
+
+  // Add a response interceptor to return the response
+  const responseInterceptor = async <TData = unknown>(
+    response: Response
+  ): Promise<FetchResponse<TData>> => {
+    const result = await response.json();
+
+    return {
+      data: result,
+    };
+  };
+
+  const paramsSerializer = <Param = unknown>(url: string, params?: Param) => {
+    if (params) {
+      const serializedParams = qs.stringify(params, { encode: false });
+      return `${url}?${serializedParams}`;
+    }
+    return url;
+  };
+
+  const addBaseUrl = (url: FetchURL, clientBaseURL?: string) => {
```

Comment:
what circumstances would we be passing a `baseUrl` that isn't the URL set in the config of the application?

Reply:
well, essentially it was possible before, it was possible to change the baseUrl for 1 method and override the main configuration, I was just attempting to preserve that behavior, but I don't know if we are making use of this anywhere else.
we can choose to remove that possibility, considering it could be added later on if we need it